haas/cli.py

```diff
@@ -236,12 +236,31 @@ def headnode_stop(headnode):
     do_post(url)
 
 @cmd
-def node_register(node, ipmi_host, ipmi_user, ipmi_pass):
-    """Register a node named <node>, with the given ipmi host/user/password"""
+def node_register(node, subtype, *args):
+    """Register a node named <node>, with the given type
+	if obm is of type: ipmi then provide arguments
+	"ipmi", <hostname>, <ipmi-username>, <ipmi-password> 
+    """
+    obm_api = "http://schema.massopencloud.org/haas/v0/obm/"
+
+    if subtype=="ipmi":
+	if len(args) == 3:
+	    obminfo = {"type": obm_api+subtype, "host": args[0],
+	    		"user": args[1], "password": args[2]
+	    	      }
+	else:
+	    print "Error: subtype <ipmi> requires exactly 3 arguments"
+	    print "<hostname> <ipmi-username> <ipmi-password>"
```

Comment:
If you're going to write out custom error messages, there are a couple things to keep in mind:

1. The other error messages get printed to stderr, so these should do the same.
2. It might make more sense to raise a `TypeError`, which will trigger the usual error printing mechanism (and print the docstring).

The only disadvantage of (2) is that it won't currently actually print out your custom error, but if your docstring is clear enough this might be okay. It also wouldn't be too hard to change, but given that we want to swap the whole thing out for something that uses the `argparse` library, it may not be worth bothering.

I'm fine not raising an exception and printing the error ourselves if that's what you want to do, but if you do that you should use `os.stderr.write` instead of `print`.